### PR TITLE
fix: Update WalletConnectConnector to handle individual namespaces like eip155:*

### PR DESCRIPTION
**Pull request by 0xAsimetriq**
## Description

Up until now WalletConnectConnector was set up to handle namespaces that included all data under global `eip155` key i.e.
```json
{
  "namespaces": {
    "eip155": {
      "chains": ["eip155:14", "eip155:19"],
      "methods": ["eth_sendTransaction", "eth_signTransaction", "personal_sign"],
      "events": ["accountsChanged", "chainChanged"],
      "accounts": ["eip155:14:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb", "eip155:19:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"]
    },
  }
}
```

This worked as that's how majority of wallets implement the spec, however with v1 sunset we saw few wallets that return namespaces in another format (also valid)

```json
{
  "namespaces": {
    "eip155:14": {
      "methods": ["eth_sendTransaction", "eth_signTransaction", "personal_sign"],
      "events": ["accountsChanged", "chainChanged"],
      "accounts": ["eip155:14:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"]
    },
    "eip155:19": {
      "methods": ["eth_sendTransaction", "eth_signTransaction", "personal_sign"],
      "events": ["accountsChanged", "chainChanged"],
      "accounts": ["eip155:19:0xab16a96d359ec26a11e2c2b3d8f8b8942d5bfcdb"]
    },
  }
}
```

Plus, there can be a combination of both formats returned. I updated connector to handle both when checking namespace chainId's and methods for validation.

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/references/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address: asimetriq.eth

### CODE REVIEW

File: packages/connectors/package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "@safe-global/safe-apps-provider": "^0.17.1",
     "@safe-global/safe-apps-sdk": "^8.0.0",
     "@walletconnect/ethereum-provider": "2.8.6",
+    "@walletconnect/utils": "2.8.6",
```
Comment (tmm) [inline packages/connectors/package.json:29]: Can we add `sideEffects: false` to [`@walletconnect/utils#package.json`](https://github.com/WalletConnect/walletconnect-monorepo/blob/v2.0/packages/utils/package.json)

Comment (0xAsimetriq) [reply packages/connectors/package.json:29]: Created pr for this, but not sure if we will land it this week
https://github.com/WalletConnect/walletconnect-monorepo/pull/2910

Comment (ganchoradkov) [reply packages/connectors/package.json:29]: @0xAsimetriq aren't the utils pre-bundled with `ethereum-provider`? with yarn I'm sure it works without specifying the package, not sure about pnpm tho

**Comment by midotrinh1211** [comment]
This PR seems to be fixing for this bug report:
https://github.com/wagmi-dev/references/issues/392

Pls review/merge it asap

Thanks

**Comment by 0xAsimetriq** [comment]
Pr is out of sync with actual branch will close and reopen to try and get latest updates. Looks like gh bug

**Comment by midotrinh1211** [comment]
@0xAsimetriq 
For the `wagmi` versions after the version `1.3.4`, for example version `1.3.5` or `1.3.7`, we here even encounter the below error when starting the dApp:
`error - Error: `useConfig` must be used within `WagmiConfig`` 

The error is related to the usage of the hook `useNetwork`

Pls notice that this error does not occur for wagmi version `1.3.4`

**Comment by 0xAsimetriq** [comment]
@midotrinh1211 

1. Maybe website wasn't updated yet or there is an issue with Bifrost
2. Make sure you use wagmi hooks inside `<WagmiConfig />` component with correct setup

**Comment by midotrinh1211** [comment]
> @midotrinh1211
> 
> 1. Maybe website wasn't updated yet or there is an issue with Bifrost
> 2. Make sure you use wagmi hooks inside `<WagmiConfig />` component with correct setup

@0xAsimetriq 
Thanks a lot for your reply.

Regarding point #1, if possible, pls discuss directly with Bifrost as it is the main mobile wallet on Songbird/Flare blockchain networks.

Regarding point #2, that error does not occur for `wagmi` version `1.3.4`. The error occurs only for versions `1.3.5` and `1.3.7`
Looking forward to your further reply on this.

**Comment by midotrinh1211** [comment]
@0xAsimetriq 
With the latest version of `1.3.8`, still same issue as reported above.

**Pull request closed without merging**